packages/superset-ui-encodable/src/fillers/fillAxisConfig.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-magic-numbers */
+import isEnabled from '../utils/isEnabled';
+import { isTypedFieldDef } from '../typeGuards/ChannelDef';
+import { ChannelDef, PositionFieldDef } from '../types/ChannelDef';
+import { ChannelType } from '../types/Channel';
+import { isXY, isX } from '../typeGuards/Channel';
+import { RequiredSome } from '../types/Base';
+import { AxisConfig, LabelOverlapStrategy } from '../types/Axis';
+import expandLabelOverlapStrategy from './expandLabelOverlapStrategy';
+
+function isChannelDefWithAxisSupport(
+  channelType: ChannelType,
+  channelDef: ChannelDef,
+): channelDef is PositionFieldDef {
+  return isTypedFieldDef(channelDef) && isXY(channelType);
+}
+
+export type FilledAxisConfig =
```

Comment:
`CompleteAxisConfig` is a little more clear to me than `FilledXXX`.

Wonder if we could use `complete` everywhere instead of `fill`, there's a bit of variation between files, e.g., `completeAxisConfig`, `completeChannelDef`, etc.

Reply:
sgtm